pyvista/plotting/scalar_bars.py

```diff
@@ -28,12 +28,34 @@ def __repr__(self):
         """Nice representation of this class."""
         lines = []
         lines.append('Scalar Bar Title     Interactive')
-        for title, actor in self._scalar_bar_actors.items():
+        for title in self._scalar_bar_actors:
             interactive = title in self._scalar_bar_widgets
             title = f'"{title}"'
             lines.append(f'{title:20} {str(interactive):5}')
         return '\n'.join(lines)
 
+    def _remove_mapper_from_plotter(self, actor, reset_camera=False, render=False):
+        """Remove an actor's mapper from the given plotter's _scalar_bar_mappers."""
+        try:
+            mapper = actor.GetMapper()
+        except AttributeError:
+            return
+        for name in list(self._scalar_bar_mappers.keys()):
+            try:
+                self._scalar_bar_mappers[name].remove(mapper)
+            except ValueError:
+                pass
+            if len(self._scalar_bar_mappers[name]) < 1:
+                slot = self._plotter._scalar_bar_slot_lookup.pop(name, None)
+                if slot is not None:
+                    self._scalar_bar_mappers.pop(name)
+                    self._scalar_bar_ranges.pop(name)
+                    self._plotter.remove_actor(self._scalar_bar_actors.pop(name),
```

Comment:
I was at first a bit wary of calling that `pop` inside the function call, because if the key is missing from the dict then the error message could be confusing. But as far as I can tell the three dicts on lines 51-53 get set hand in hand, so hopefully they can never get out of sync.